lib/shared/store.js

```diff
@@ -432,24 +443,39 @@ const storeReducer = (state = DEFAULT_STATE, action) => {
         })
       }
 
-      const selectedDrive = findDrive(state, state.getIn([ 'selection', 'drive' ]))
+      const selectedDevices = state.getIn([ 'selection', 'devices' ])
 
-      return _.attempt(() => {
-        if (selectedDrive && !_.every([
-          constraints.isDriveValid(selectedDrive.toJS(), action.data),
-          constraints.isDriveSizeRecommended(selectedDrive.toJS(), action.data)
-        ])) {
-          return storeReducer(state, {
-            type: ACTIONS.DESELECT_DRIVE
+      // Remove image-incompatible drives from selection with `constraints.isDriveValid`
+      return _.reduce(selectedDevices.toJS(), (accState, device) => {
+        const drive = _.find(getAvailableDrives(state), { device })
+        if (!constraints.isDriveValid(drive, action.data) || !constraints.isDriveSizeRecommended(drive, action.data)) {
+          return storeReducer(accState, {
+            type: ACTIONS.DESELECT_DRIVE,
+            data: device
           })
         }
 
-        return state
-      }).setIn([ 'selection', 'image' ], Immutable.fromJS(action.data))
+        return accState
+      }, state).setIn([ 'selection', 'image' ], Immutable.fromJS(action.data))
     }
 
     case ACTIONS.DESELECT_DRIVE: {
-      return state.deleteIn([ 'selection', 'drive' ])
+      if (!action.data) {
```

Comment:
For consistency with the other actions, you could also assign this `action.data` to a const variable?

Reply:
Let's not assign stuff to vars just for the sake of it; object property access is just fine, and should be preferred, unless too unweildy

Reply:
I dunno if the same is true in JavaScript-land, but I remember reading that in Python when you have performance-critical code, doing
```
foo = bar.thing
do_something(foo)
do_something_else(foo)
foo.wibble()
```
can be slightly faster than
```
do_something(bar.thing)
do_something_else(bar.thing)
bar.thing.wibble()
```
because it then does the property-lookup on `bar` just once rather than multiple times. \*shrug\*

> imo the `const something = action.data` should be removed in favor of accessing the props directly

Even if it makes the code less readable? \*shrug\* again ;)

Reply:
The reason for the `const`s above is because it makes code in the same block easier to write (object shorthand syntax), whereas here it wouldn't help as much, but the discussion of `const` performance is worth continuing in an issue.